packages/trader/src/Stores/Modules/Trading/Helpers/proposal.ts

```diff
@@ -0,0 +1,185 @@
+import { PriceProposalResponse } from '@deriv/api-types';
+import { convertToUnix, getDecimalPlaces, getPropertyValue, isAccumulatorContract, toMoment } from '@deriv/shared';
+import { useTraderStore } from 'Stores/useTraderStores';
+
+type TObjContractBasis = {
+    text: string;
+    value: string;
+};
+
+type TObjMultiplier = {
+    cancellation?: string;
+    limit_order?: {
+        take_profit?: number;
+        stop_loss?: number;
+    };
+    multiplier?: number;
+};
+
+type TObjAccum = {
+    growth_rate?: number;
+    limit_order?: {
+        take_profit?: number;
+    };
+};
+
+type TObjExpiry = {
+    date_expiry?: number;
+};
+
+const isVisible = (elem: HTMLElement) => !(!elem || (elem.offsetWidth === 0 && elem.offsetHeight === 0));
+
+const map_error_field: { [key: string]: string } = {
+    barrier: 'barrier_1',
+    barrier2: 'barrier_2',
+    date_expiry: 'expiry_date',
+};
+
+export const getProposalErrorField = (response: PriceProposalResponse) => {
+    const error_field: string = getPropertyValue(response, ['error', 'details', 'field']);
+    if (!error_field) {
+        return null;
+    }
+    const error_id = map_error_field[error_field] || error_field;
+    const el_error = document.getElementsByName(error_id)[0];
+    return el_error && isVisible(el_error) ? error_id : null;
+};
+
+export const getProposalInfo = (
+    store: ReturnType<typeof useTraderStore>,
+    response: PriceProposalResponse,
+    obj_prev_contract_basis: TObjContractBasis
+) => {
+    const proposal = response.proposal || ({} as PriceProposalResponse['proposal']);
+    const profit = (proposal?.payout || 0) - (proposal?.ask_price || 0);
+    const returns = (profit * 100) / (proposal?.ask_price || 1);
+    const stake = proposal?.display_value;
+    const basis_list = store.basis_list;
+
+    const contract_basis: TObjContractBasis | undefined = store.is_vanilla
+        ? { text: 'Payout', value: 'number_of_contracts' }
+        : basis_list.find(o => o.value !== store.basis);
+
+    const is_stake = contract_basis?.value === 'stake';
+    //@ts-expect-error we are using key that does not exist in proposal
```

Comment:
@henry-deriv please use an assertion (as) or a type guard (if conditions) in places, where you have TS complaining about the types it's not aware about, instead of this comment.